src/components/forms/SelectGPUModal.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC } from "react";
+import { Button, MainTable, Modal } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchResources } from "api/server";
+import Loader from "components/Loader";
+import ScrollableTable from "components/ScrollableTable";
+import { GpuCard } from "types/resources";
+
+interface Props {
+  onSelect: (image: GpuCard) => void;
+  onClose: () => void;
+}
+
+const SelectGPUModal: FC<Props> = ({ onSelect, onClose }) => {
+  const { data: resources, isLoading } = useQuery({
+    queryKey: [queryKeys.resources],
+    queryFn: () => fetchResources(),
+  });
+
+  const headers = [
+    { content: "Vendor" },
+    { content: "Driver" },
+    { content: "PCI address" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = isLoading
+    ? []
+    : resources?.gpu?.cards?.map((card) => {
+        const selectCard = () => onSelect(card);
+
+        return {
+          className: "u-row",
+          columns: [
+            {
+              content: card.vendor,
+              role: "cell",
+              "aria-label": "Vendor",
+              onClick: selectCard,
+            },
+            {
+              content: (
+                <>
+                  {card.driver}{" "}
+                  <span className="u-text--muted">{card.driver_version}</span>
+                </>
+              ),
+              role: "cell",
+              "aria-label": "Driver",
+              onClick: selectCard,
+            },
+            {
+              content: card.pci_address,
+              role: "cell",
+              "aria-label": "PCI Address",
+              onClick: selectCard,
+            },
+            {
+              content: (
+                <Button
+                  onClick={selectCard}
```

Comment:
do we need `selectCard` passed to both onClick for the button and the table cell?

Reply:
Yes, to click the table cell, we do the same for base image or storage volume selection. It makes the button a bit pointless, because we don't need it, but I think it serves a purpose to guide the user what to do.